### PR TITLE
Add regression test for #5982

### LINKED ISSUE #5982: TypeError: '<' not supported between instances of 'int' and 'NoneType'
### Bug description

# Steps to reproduce

1. Create python file `test.py` with contents:

```python
from __future__ import annotations

from collections import namedtuple


class SomeClass(namedtuple("SomeClass", [])):
    def method(self) -> SomeClass:
        pass
```

2. Run pylint against it:

```
$ pylint -d C0114,C0115,C0116 test.py
```

# Expected

 - pylint succeeds
 - worked as expected with `pylint==2.12.2` and earlier versions of pylint

# Actual

```
Exception on node <Name.SomeClass l.7 at 0x112389a10> in file '/private/tmp/pylintfail/crash_pylint.py'
Traceback (most recent call last):
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 73, in walk
    callback(astroid)
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1362, in visit_name
    self._undefined_and_used_before_checker(node, stmt)
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1405, in _undefined_and_used_before_checker
    node, stmt, frame, current_consumer, i, base_scope_type
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1595, in _check_consumer
    is_recursive_klass,
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1866, in _is_variable_violation
    maybe_before_assign = _detect_global_scope(node, frame, defframe)
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 262, in _detect_global_scope
    return frame.lineno < defframe.lineno
TypeError: '<' not supported between instances of 'int' and 'NoneType'
************* Module crash_pylint
crash_pylint.py:1:0: F0001: Fatal error while checking 'crash_pylint.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/Users/name/Library/Caches/pylint/pylint-crash-2022-03-25-14.txt'. (fatal)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Configuration

_No response_

### Command used

```shell
pylint -d C0114,C0115,C0116 test.py
```

### Pylint output

```shell
Exception on node <Name.SomeClass l.7 at 0x112389a10> in file '/private/tmp/pylintfail/crash_pylint.py'
Traceback (most recent call last):
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 73, in walk
    callback(astroid)
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1362, in visit_name
    self._undefined_and_used_before_checker(node, stmt)
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1405, in _undefined_and_used_before_checker
    node, stmt, frame, current_consumer, i, base_scope_type
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1595, in _check_consumer
    is_recursive_klass,
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1866, in _is_variable_violation
    maybe_before_assign = _detect_global_scope(node, frame, defframe)
  File "/private/tmp/pylintfail/.venv/lib/python3.7/site-packages/pylint/checkers/variables.py", line 262, in _detect_global_scope
    return frame.lineno < defframe.lineno
TypeError: '<' not supported between instances of 'int' and 'NoneType'
************* Module crash_pylint
crash_pylint.py:1:0: F0001: Fatal error while checking 'crash_pylint.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/Users/name/Library/Caches/pylint/pylint-crash-2022-03-25-14.txt'. (fatal)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

pylint succeeds

### Pylint version

```shell
pylint 2.13.0
astroid 2.11.1
Python 3.7.10 (default, Apr  2 2021, 10:55:19) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #5982
Refs PyCQA/astroid#1489

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Hmm, there's a fail in 3.6, but it's end of life. We're effectively maintaining 3.6 code as long as we do not release 2.14 with https://github.com/PyCQA/pylint/pull/5921. It may make sense to release 2.14 now with the few features it has in order to not maintain python 3.6 any longer.

File: tests/functional/u/used/used_before_assignment_type_annotations.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 """Tests for annotation of variables and potential use before assignment"""
 # pylint: disable=too-few-public-methods, global-variable-not-assigned
+from __future__ import annotations
```
Comment (cdce8p) [inline tests/functional/u/used/used_before_assignment_type_annotations.py:3]: The `__future__` import was adding in Python 3.7. I would suggest setting that as the min Python version for the test. No need to test 3.6 separately.

**Comment by jacobtylerwalls** [comment]
Ha, no I just need to set min version on the test -- forgot that `from __future__ import annotations` doesn't exist in py3.6. I'll push.

File: tests/functional/u/used/used_before_assignment_type_annotations.py (hunk 1 above)
Comment (jacobtylerwalls) [reply tests/functional/u/used/used_before_assignment_type_annotations.py:3]: jinx!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by DanielNoord** [comment]
Please see comment in `astroid` PR. I'm not sure if the fix there is correct and we might want to hold of on merging this until discussion has resolved there.

**Comment by Pierre-Sassoulas** [comment]
Merging as the discussion in https://github.com/PyCQA/astroid/pull/1489#issuecomment-1079788213 came to an agreement that the fix is correct.

**Pull request merged**